packages/cdk/lib/transcription-service.ts

```diff
@@ -67,5 +92,92 @@ export class TranscriptionService extends GuStack {
 			ttl: Duration.hours(1),
 			resourceRecord: apiDomain.domainNameAliasDomainName,
 		});
+
+		// worker autoscaling group
+
+		const workerApp = `${APP_NAME}-worker`;
+		const userData = UserData.forLinux({ shebang: '#!/bin/bash' });
+		// basic placeholder commands
+		userData.addCommands(
+			[
+				`aws s3 cp s3://${GuDistributionBucketParameter.getInstance(this).valueAsString}/${props.stack}/${props.stage}/${workerApp}/worker.zip .`,
+				`unzip worker.zip`,
+				`node index.js`,
+			].join('\n'),
+		);
+
+		const role = new GuInstanceRole(this, {
+			app: workerApp,
+		});
+
+		const launchTemplate = new LaunchTemplate(
+			this,
+			'TranscriptionWorkerLaunchTemplate',
+			{
+				machineImage: MachineImage.genericLinux({
+					'eu-west-1': workerAmi.valueAsString,
+				}),
+				instanceType: InstanceType.of(InstanceClass.C7G, InstanceSize.XLARGE4),
+				// the size of this block device will determine the max input file size for transcription. In future we could
+				// attach the block device on startup once we know how large the file to be transcribed is, or try some kind
+				// of streaming approach to the transcription so we don't need the whole file on disk
+				blockDevices: [
+					{
+						deviceName: '/dev/sda1',
+						// assuming that we intend to support video files, 50GB seems a reasonable starting point
+						volume: BlockDeviceVolume.ebs(50),
+					},
+				],
+				userData,
+				role: role,
+			},
+		);
+
+		// instance types we are happy to use for workers. Note - order matters as when launching 'on demand' instances
+		// the ASG will start at the top of the list and work down until it manages to launch an instance
+		const acceptableInstanceTypes = isProd
+			? [
+					InstanceType.of(InstanceClass.C7G, InstanceSize.XLARGE4),
+					InstanceType.of(InstanceClass.C6G, InstanceSize.XLARGE4),
+					InstanceType.of(InstanceClass.M7G, InstanceSize.XLARGE4),
+					InstanceType.of(InstanceClass.C7G, InstanceSize.XLARGE8),
+					InstanceType.of(InstanceClass.C6G, InstanceSize.XLARGE8),
+				]
+			: [InstanceType.of(InstanceClass.T4G, InstanceSize.MEDIUM)];
+
+		// unfortunately GuAutoscalingGroup doesn't support having a mixedInstancesPolicy so using the basic ASG here
+		new AutoScalingGroup(this, 'TranscriptionWorkerASG', {
+			minCapacity: 0,
+			maxCapacity: isProd ? 20 : 4,
+			autoScalingGroupName: `transcription-service-workers-${this.stage}`,
+			vpc: GuVpc.fromIdParameter(this, 'InvestigationsInternetEnabledVpc', {
+				availabilityZones: ['eu-west-1a', 'eu-west-1b', 'eu-west-1c'],
```

Comment:
How does this know what VPC to choose from? Is there supposed to be a parameter in the stack named `InvestigationsInternetEnabledVpc`?

Reply:
This is a bit of GuCDK magic - behind the scenes it looks for a parameter called /account/primary/vpc/id in parameter store. `InvestigationsInternetEnabledVpc` is just a cdk id here - it is not used for anything

Reply:
Thanks for explaining 👍